akka-http/src/main/scala/akka/http/scaladsl/server/util/JwtSupport.scala

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2022 Lightbend Inc.
+ */
+
+package akka.http.scaladsl.server.util
+
+import java.io.File
+import java.security.KeyPair
+import java.util.Base64
+
+import com.typesafe.config.Config
+import javax.crypto.SecretKey
+import javax.crypto.spec.SecretKeySpec
+import pdi.jwt.{ algorithms, JwtAlgorithm, JwtOptions }
+import spray.json.{ JsObject, JsString }
+
+import scala.util.{ Failure, Success, Try }
+import scala.jdk.CollectionConverters._
+
+trait JwtSupport {
+
+  def canValidate: Boolean
+
+  def validate(token: String): Either[Exception, JsObject]
+
+}
+
+object JwtSupport {
+
+  private val NoValidationOptions = JwtOptions.DEFAULT.copy(signature = false, expiration = false, notBefore = false)
+
+  def fromConfig(config: Config): JwtSupport = {
+    val jwtConfig = config.getConfig("akka.http.jwt")
+    val devSecret = if (jwtConfig.getBoolean("dev")) {
+      Some(JwtSecret("dev", Some(jwtConfig.getString("dev-issuer")).filter(_.nonEmpty), JwtNoneAlgorithmSecret))
+    } else None
+
+    val secrets = jwtConfig
+      .getConfigList("secrets")
+      .asScala
+      .map { secretConfig =>
+        val keyId = secretConfig.getString("key-id")
+        val algorithmSecret: JwtAlgorithmSecret = secretConfig.getString("algorithm") match {
+          case "none" => JwtNoneAlgorithmSecret
+          case alg =>
+            val algorithm = JwtAlgorithm.fromString(alg)
+            if (secretConfig.hasPath("secret-dir")) {
+              val dir = new File(secretConfig.getString("secret-dir"))
```

Comment:
Pick up all/both the discrete paths from config rather than directory from config with hardcoded magic filenames.